api/proquest/importer.py

```diff
@@ -789,61 +811,199 @@ def _parse_identifier(self, identifier):
+            for group in feed.groups:
+                if group.publications:
+                    for publication in group.publications:
+                        yield publication
+
+    def _download_feed_pages(self):
+        """Download all the pages of the ProQuest OPDS feed.
 
-        feeds = []
+        Downloaded feed pages are dumped to the local disk instead of storing them in memory
+        in order to decrease the memory footprint.
+
+        :return: List of temporary file handles pointing to feed pages
+        :rtype: List[file]
+        """
+        feed_temporary_files = []
 
         self._logger.info("Started downloading feed pages")
 
         for feed in self._client.download_all_feed_pages(self._db):
-            feeds.append(feed)
+            feed_temporary_file = tempfile.TemporaryFile(mode="r+")
+            feed_temporary_file.write(json.dumps(feed))
+            feed_temporary_file.flush()
 
-        self._logger.info("Finished downloading {0} feed pages".format(len(feeds)))
+            feed_temporary_files.append(feed_temporary_file)
 
-        page = 1
-        processed_number_of_items = 0
-        total_number_of_items = None
+        self._logger.info(
+            "Finished downloading {0} feed pages".format(len(feed_temporary_files))
+        )
+
+        return feed_temporary_files
+
+    def _parse_feed(self, page, feed_temporary_file):
+        """Parse the ProQuest feed page residing in a temporary file on a local disk.
+
+        :param page: Index of the page
+        :type page: int
+
+        :param feed_temporary_file: File handle pointing to a temporary file containing the feed page
+        :type feed_temporary_file: file
+
+        :return: Parsed OPDS feed page
+        :rtype: opds2_ast.OPDS2Feed
+        """
+        self._logger.info("Page # {0}. Started parsing the feed".format(page))
+
+        feed_temporary_file.seek(0)
+        feed = feed_temporary_file.read()
+        feed = parse_feed(feed, silent=False)
+
+        self._logger.info("Page # {0}. Finished parsing the feed".format(page))
 
-        self._logger.info("Started processing feed pages")
+        return feed
 
-        for feed in feeds:
-            self._logger.info("Page # {0}. Started parsing the feed".format(page))
+    def _collect_feed_identifiers(self, feed, feed_identifiers):
+        """Keep track of all identifiers in the ProQuest feed and save them in a list.
 
-            feed = parse_feed(feed, silent=False)
+        :param feed: ProQuest OPDS 2.0 feed
+        :type feed: opds2_ast.OPDS2Feed
 
-            self._logger.info("Page # {0}. Finished parsing the feed".format(page))
+        :param feed_identifiers: List of identifiers in the ProQuest feed
+        :type feed_identifiers: List[str]
+        """
+        for publication in self._get_publications(feed):
+            identifier = parse_identifier(self._db, publication.metadata.identifier)
+
+            feed_identifiers.append(identifier.identifier)
```

Comment:
Don't bother changing this, but if this method returned its own list of identifiers, rather than modifying a list in place, it wouldn't need to take `feed_identifiers` as an argument.